registry/auth/token/token.go

```diff
@@ -204,19 +208,20 @@ func (t *Token) VerifySigningKey(verifyOpts VerifyOptions) (signingKey libtrust.
 
 	switch {
 	case len(x5c) > 0:
-		signingKey, err = parseAndVerifyCertChain(x5c, verifyOpts.Roots)
+		signingKey, err := parseAndVerifyCertChain(x5c, verifyOpts.Roots)
+		return []libtrust.PublicKey{signingKey}, err
 	case rawJWK != nil:
-		signingKey, err = parseAndVerifyRawJWK(rawJWK, verifyOpts)
+		signingKey, err := parseAndVerifyRawJWK(rawJWK, verifyOpts)
+		return []libtrust.PublicKey{signingKey}, err
 	case len(keyID) > 0:
```

Comment:
As far as I understand, the JWK Key ID is used in the JWKS URI from OpenID, where multiple keys might be specified (see https://code.beryju.org/oauth/discovery/keys), and in that case the Key ID used in that document needs to match the Key ID in the JWT, as well as signature being validated with that same key.

Since this project directly compares against a list of certificates instead of JWK-formatted key entries, there's no way to pre-specify the key ID since it is not encoded in the certificate in any way

I also don't think this will allow the loading of keys from different certificates, since a JWT is always signed by a single public/private key pair, and there is no certificate chain embedded into the signature, so given certificateA is issued by intermediateA, and a JWT signed by certificateA, it will only be validated with the certificateA and not intermediateA

Reply:
> since a JWT is always signed by a single public/private key pair, and there is no certificate chain embedded into the signature, so given certificateA is issued by intermediateA, and a JWT signed by certificateA, it will only be validated with the certificateA and not intermediateA

[That is incorrect.](https://datatracker.ietf.org/doc/html/rfc7515#section-4.1.6) JSON Web Signatures can and do embed the certificate chain so that tokens signed by certificateA can be verified using the CA certificate which issued intermediateA.